crates/sui-core/src/gateway_state.rs

```diff
@@ -693,16 +693,25 @@ where
 
     async fn download_object_from_authorities(&self, object_id: ObjectID) -> SuiResult<ObjectRead> {
         let result = self.authorities.get_object_info_execute(object_id).await?;
-        if let ObjectRead::Exists(obj_ref, object, _) = &result {
-            let local_object = self.store.get_object(&object_id)?;
-            if local_object.is_none()
-                // We only update local object if the validator version is newer.
-                || local_object.unwrap().version() < obj_ref.1
-            {
-                self.store.insert_object_direct(*obj_ref, object).await?;
+        match &result {
+            ObjectRead::Exists(object_ref, object, _) => {
+                debug!(?object_ref, "Download object result: object exists");
+                let local_object = self.store.get_object(&object_id)?;
+                if local_object.is_none()
+                    // We only update local object if the validator version is newer.
+                    || local_object.unwrap().version() < object_ref.1
+                {
+                    self.store.insert_object_direct(*object_ref, object).await?;
+                }
+                trace!(?object_id, "Downloaded object details: {}", result);
+            }
+            ObjectRead::Deleted(object_ref) => {
+                debug!(?object_ref, "Download object result: object is deleted");
+            }
+            ObjectRead::NotExists(_) => {
+                debug!(?object_id, "Download object result: object does not exist");
```

Comment:
This seems to just make it more noisy?

Reply:
We already have: https://github.com/MystenLabs/sui/blob/87d5950c06dfc8cd8931ffe89ebcce4cd8c9060d/crates/sui-types/src/object.rs#L620

Reply:
No, it's less. 

We use to `debug!` for either of the three cases. And the thing we dislike the most is when it exists. With this change, we still `debug!` for all cases but when it exists it prints much fewer things

Reply:
Maybe I am missing something, but it's supposed to only print the object ref when the object exists (https://github.com/MystenLabs/sui/blob/87d5950c06dfc8cd8931ffe89ebcce4cd8c9060d/crates/sui-types/src/object.rs#L630)

Reply:
Note that we used to print with "{:?}" which is why it's displaying the content.
https://github.com/MystenLabs/sui/pull/3358 fixed it and now it's "{}" which will call the Display implementation.

Reply:
Looking at the code, I believe @lxfind is right.  The `{}` of `ObjectRead` should print out only the ref, and not the contents. Thus for object exists, the change makes no real difference, other than that at `trace` level we can get the details.